### PR TITLE
docs: Update README.md with revised information

### PR DESCRIPTION
**Pull request by rojandahal**
The documentation (https://github.com/fastify/fastify-oauth2/blob/master/README.md?plain=1#L230) explained that the refresh token required is actual refresh token. 

But the refresh token passed to `simple-oauth2` 's `.createToken()` method expects the Access Token object with refresh token itself.

So the documentation is edited to explain that the `.getNewAccessTokenUsingRefreshToken()` method required AccessToken object which is of format `{refresh_token: 'eyJhbGciOiJSUzI1NiI...'})`

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by Eomm — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -227,7 +227,13 @@ This fastify plugin adds 3 utility decorators to your fastify instance using the
...
+- `getNewAccessTokenUsingRefreshToken(AccessToken, params, callback)`: A function that takes a AccessToken and retrieves a new `AccessToken` object. This is generally useful with background processing workers to re-issue a new token when the original token has expired. The `params` argument is optional and it is an object that can be used to pass in extra parameters to the refresh request (e.g. a stricter set of scopes). If the callback is not passed this function will return a promise. The object resulting from the callback call or the promise resolution is a new `AccessToken` object (see above). Example of how you would use it for `name:googleOAuth2`:
```
Comment (Uzlopak) [inline README.md:230]: This does not look right.

Should be more like:

```suggestion
- `getNewAccessTokenUsingRefreshToken(Token, params, callback)`: A function that takes a `Token`-Object and retrieves a new `Token`-Object. This is generally useful with background processing workers to re-issue a new AccessToken when the previous AccessToken has expired. The `params` argument is optional and it is an object that can be used to pass in additional parameters to the refresh request (e.g. a stricter set of scopes). If the callback is not passed this function will return a Promise. The object resulting from the callback call or the resolved Promise is a new `Token` object (see above). Example of how you would use it for `name:googleOAuth2`:
```

**Review by Uzlopak — CHANGES_REQUESTED** [review]
Does not seem right

File: README.md (hunk 1 above)
Comment (rojandahal) [reply README.md:230]: The `Token` - object is actually an AccessToken object as mentioned here: [AccessToken](https://github.com/lelylan/simple-oauth2/blob/master/API.md#accesstoken).  `getNewAccessTokenUsingRefreshToken(Token, params, callback)` is okay. Other changes can be made as `AccessToken`-Object

```suggestion
- `getNewAccessTokenUsingRefreshToken(Token, params, callback)`: A function that takes a `AccessToken`-Object as `Token` and retrieves a new `AccessToken`-Object. This is generally useful with background processing workers to re-issue a new AccessToken when the previous AccessToken has expired. The `params` argument is optional and it is an object that can be used to pass in additional parameters to the refresh request (e.g. a stricter set of scopes). If the callback is not passed this function will return a Promise. The object resulting from the callback call or the resolved Promise is a new `AccessToken` object (see above). Example of how you would use it for `name:googleOAuth2`:
```

Comment (rojandahal) [reply README.md:230]: @Uzlopak Does the new suggested change looks right ?

**Review by Uzlopak — APPROVED** [review]
LGTM. :)

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**